src/huggingface_hub/file_download.py

```diff
@@ -1748,7 +1748,12 @@ def _get_metadata_or_catch_error(
                     if commit_hash is not None:
                         no_exist_file_path = Path(storage_folder) / ".no_exist" / commit_hash / relative_filename
                         no_exist_file_path.parent.mkdir(parents=True, exist_ok=True)
-                        no_exist_file_path.touch()
+                        try:
+                            no_exist_file_path.touch()
+                        except OSError as e:
+                            logger.error(
+                                f"Could not cache non-existence of file. Will ignore error and continue. Error: {e}"
+                            )
```

Comment:
The current issue we have is that we can't touch (with S3) if this file exists.

This change here solves that issue (I confirmed with CI), but it's more general (i.e. whenever there is an OSSError no matter what cuases it).

Should we check if the file exists anyway just after this block?

Reply:
No I don't think so. As I mentioned in the PR description, this file exists only to save a HEAD call so it's a very optional optimization. If it fails because of a filesystem-specific issue then it's not a problem.

Reply:
thanks!